File: login/src/main/java/karthee/login/db/remote/APIService.java
package karthee.login.db.remote;


import io.reactivex.Observable;
import karthee.login.model.LoginRequest;
import karthee.login.model.LoginResponse;
import retrofit2.http.Body;
import retrofit2.http.POST;


public interface APIService {

    String BASE_URL = "http://demo6183250.mockable.io/";    // mocked login service

    @POST("login")   // mocked login service
    Observable<LoginResponse> login(@Body LoginRequest request);
}

File: login/src/main/java/karthee/login/db/remote/APIService.kt
package karthee.login.db.remote


import io.reactivex.Observable
import karthee.login.model.LoginRequest
import karthee.login.model.User
import retrofit2.http.Body
import retrofit2.http.POST


interface APIService {
    @POST("Signin")   // mocked login service
    fun login(@Body request: LoginRequest): Observable<User>

    companion object {

        val BASE_URL = "http://52.90.34.153:92/"    // mocked login service
    }
}
